engine/src/cmd/weapon_factory.cpp

```diff
@@ -0,0 +1,146 @@
+        wi.size = getMountSize(inner.get( "<xmlattr>.mountsize", "Unknown_mount"));
+
+        // Energy
+        wi.energy_rate = inner.get( "Energy.<xmlattr>.rate", wi.energy_rate );
+        wi.stability = inner.get( "Energy.<xmlattr>.stability", wi.stability );
+        wi.refire_rate = inner.get( "Energy.<xmlattr>.refire", wi.refire_rate );
+        wi.lock_time = inner.get( "Energy.<xmlattr>.locktime", wi.lock_time );
+
+        // Damage
+        // TODO: weapon_list.xml laser has damage. Everything else has rate. Correct.
+        wi.damage = inner.get( "Damage.<xmlattr>.rate", wi.damage );
+        wi.phase_damage = inner.get( "Damage.<xmlattr>.phasedamage", wi.phase_damage );
+        wi.radius = inner.get( "Damage.<xmlattr>.radius", wi.radius );
+        wi.radial_speed = inner.get( "Damage.<xmlattr>.radialspeed", wi.radial_speed );
+        wi.long_range = inner.get( "Damage.<xmlattr>.longrange", wi.long_range );
+
+        // Distance
+        wi.volume = inner.get( "Distance.<xmlattr>.volume", wi.volume );
+        wi.speed = inner.get( "Distance.<xmlattr>.speed", wi.speed );
+        wi.pulse_speed = inner.get( "Distance.<xmlattr>.pulsespeed", wi.pulse_speed );
+        wi.range = inner.get( "Distance.<xmlattr>.range", wi.range );
+        wi.length = inner.get( "Distance.<xmlattr>.length", wi.length );
+
+        // TODO: this is a bug. It gets parsed into the same radius and radial_speed variables.
+        wi.radius = inner.get( "Distance.<xmlattr>.radius", wi.radius );
+        wi.radial_speed = inner.get( "Distance.<xmlattr>.radialspeed", wi.radial_speed );
+
+        // TODO: detonation range not implemented. Incorrectly assigns to pulse_speed...
+        //wi.bug = inner.get( "Energy.<xmlattr>.detonationrange", wi.bug );
+
+        // TODO: is this really necessary???
+        /*if(game_options.gun_speed_adjusted_game_speed) {
+            if (wi.speed < 1000) {
+                wi.speed *= 1.0+gun_speed/1.25;
+            } else if (wi.speed < 2000) {
+                wi.speed *= 1.0+gun_speed/2.5;
+            } else if (wi.speed < 4000) {
+                wi.speed *= 1.0+gun_speed/6.0;
+            } else if (wi.speed < 8000) {
+                wi.speed *= 1.0+gun_speed/17.0;
+            }
+        }*/
+
+        // Appearance
+        wi.file = inner.get( "Appearance.<xmlattr>.file", wi.file );
+        wi.a = inner.get( "Appearance.<xmlattr>.a", wi.a );
+        wi.r = inner.get( "Appearance.<xmlattr>.r", wi.r );
+        wi.g = inner.get( "Appearance.<xmlattr>.g", wi.g );
+        wi.b = inner.get( "Appearance.<xmlattr>.b", wi.b );
+        wi.a = inner.get( "Appearance.<xmlattr>.a", wi.a );
+
+        // TODO: reenable? No idea why or what this is...
+        /*if ( (gamma > 0) && gamma_needed( gamma, counts, 32 ) ) {
+            //approximate the color func
+            wi.b = (wi.b+color_step*5)/255.;
+            wi.g = (wi.g+color_step/5)/255.;
+            wi.r = (wi.r+color_step*2)/255.;
+        }*/
```

Comment:
looks like something to do with Gamma color correction, but don't know why. Seems like that would be better for a shader to do.

Reply:
Replaced all printf with boost but left one VSFileSystem::vs_fprintf. 
Regarding case sensitive, I'm nitpicking myself to death. The code is good enough for now and better than it was. I or someone else will return to it after we refactor the weapons list. 
Regarding the gamma, I left it as a comment because of the fence.  There's more commented code at line 100 or so. If we ever decide to revisit this and not delete it at some point, I'd want a code that makes some sense and doesn't have random integers based on one of the elder devs whims.